github/resource_github_user_gpg_key.go

```diff
@@ -51,11 +53,18 @@ func resourceGithubUserGpgKeyRead(d *schema.ResourceData, meta interface{}) erro
 		return unconvertibleIdErr(d.Id(), err)
 	}
 
+	log.Printf("[DEBUG] Reading user GPG key: %s", d.Id())
 	key, _, err := client.Users.GetGPGKey(context.TODO(), id)
 	if err != nil {
-		log.Printf("[WARN] GitHub User GPG Key (%s) not found, removing from state", d.Id())
-		d.SetId("")
-		return nil
+		if err, ok := err.(*github.ErrorResponse); ok {
```

Comment:
this pattern is repeated so many times I wonder if it's worth a helper? Only a suggestion not an issue at all.

Reply:
I'm torn on that... I see your point, but at the same time it wouldn't save a single LOC, just a few characters 🤔 

We still need both variables, so effectively you'd end up with something like

```go
if err, ok := tryCastingToGithubError(err); ok {
```
the logic we'd decouple would be just attempt to cast the error, so I see very little benefit in decoupling it.